proptest-state-machine/src/strategy.rs

```diff
@@ -606,7 +617,7 @@ impl<
             }
             Some(InitialState) => {
                 if self.initial_state.complicate()
-                    && self.check_acceptable(None)
```

Comment:
This is the bug (in my opinion). Previously, we checked whether the **last** valid state was acceptable but when we are shrinking the `initial_state`, we need to check whether the current one is acceptable.